design-documents/graph-ql/coverage/gift-registry.graphqls

```diff
@@ -0,0 +1,278 @@
+type Customer {
+    gitft_registry_list: [GiftRegistry]
+    gitft_registry(id: ID!): GiftRegistry
+}
+
+type Query {
+    giftRegistryTypes: [GiftRegistryType] @doc(description: "Get a list of available gift registry types")
+    giftRegistrySearch(
+        registrant_firstname: String!,
+        registrant_lastname: String!,
+        gift_registry_type_id: ID,
+        searchable_dynamic_attributes: [GiftRegistryDynamicAttributeInput] @doc(description: "For select attributes ID should be provided expected. For range search, '_from' and '_to' suffixes can be added to the attribute code. For text attributes provide value for exact matching.")
+    ): [GiftRegistry] @doc(description: "Gift registry search by registrant name and additional searchable attributes.")
+}
+
+type Mutation {
+    # All mutations below should only be accessible to the registry owner
+    createGiftRegistry(gift_registry: CreateGiftRegistryInput): CreateGiftRegistryOutput
+    updateGiftRegistry(id: ID!, gift_registry: UpdateGiftRegistryInput): UpdateGiftRegistryOutput
+    addProductsToGiftRegistry(gift_registry_id: ID!, items: [GiftRegistryItemInput!]!): AddProductsToGiftRegistryOutput
+}
+
+input GiftRegistryItemInput {
+    sku: String!
+    quantity: Float!
+    parent_sku: String,
+    note: String,
+    # see https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/add-items-to-cart-single-mutation.md
+    selected_options: [String!]
+    # see https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/add-items-to-cart-single-mutation.md
+    entered_options: [EnteredOptionInput!]
+}
+
+input UpdateGiftRegistryInput {
+    event_name: String
+    type_id: String
+    message: String
+    privacy_settings: GiftRegistryPrivacySettings
+    status: GiftRegistryStatus
+    registrants: [UpdateGiftRegistryRegistrantInput!] @doc(description: "As a result of the update, only the records of provided registrants will be affected. If the registrant is missing in the request, its attributes will not be changed")
+    shipping_address: GiftRegistryShippingAddressInput
+    dynamic_attributes: [GiftRegistryDynamicAttributeInput!] @doc(description: "As a result of the update, only the values of provided attributes will be affected. If the attribute is missing in the request, its value will not be changed")
+}
+
+input CreateGiftRegistryInput {
+    event_name: String!
+    type_id: String!
+    message: String!
+    privacy_settings: GiftRegistryPrivacySettings!
+    status: GiftRegistryStatus!
+    registrants: [CreateGiftRegistryRegistrantInput!]!
+    shipping_address: GiftRegistryShippingAddressInput
+    dynamic_attributes: [GiftRegistryDynamicAttributeInput]
+}
+
+type GiftRegistryShippingAddressInput @doc(description: "Either address data or address ID should be provided") {
```

Comment:
The comment says `Either address data or address ID should be provided`, but this schema allows both to be passed. Seems like undefined behavior we should be covering

Reply:
The "should be provided" comment is also a bit confusing, because consumers are _selecting_ from this type

Reply:
How would you improve it?

Also, the way it is done now is consistent with current cart schema.

Reply:
> How would you improve it?

Very fair question! Will follow-up

Reply:
(Side-note: Think you meant for `GiftRegistryShippingAddressInput` to be an `input` rather than an ObjectType definition.

Reply:
I think only one should be provided. Validation error should happen if both are provided. Will improve documentation and mention about validation exception.